File: patch_rc.py
import json
import os

path = r"C:\Users\AnilTurkmayali\Downloads\Clubs\data_fighters.json"
try:
    with open(path, 'r', encoding='utf-8') as f:
        data = json.load(f)
        
    for fighter in data:
        if 'contract' in fighter and 'release_clause' in fighter['contract']:
            fighter['contract']['release_clause'] *= 3
            
    with open(path, 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=4)
    print("Successfully updated release clauses using Python.")
except Exception as e:
    print(f"Error: {e}")

File: patch_rc.js
const fs = require('fs');
const path = require('path');

const targetPath = path.resolve('C:/Users/AnilTurkmayali/Downloads/Clubs/data_fighters.json');

try {
    let data = fs.readFileSync(targetPath, 'utf8');
    // Using a regex to find all "release_clause": <number> and multiply by 3
    let updatedData = data.replace(/"release_clause":\s*(\d+)/g, (match, p1) => {
        let oldVal = parseInt(p1);
        let newVal = oldVal * 3; // Increase by 3x to match the new x10 / x8 scaling vs the old x3 / x4 scaling
        return `"release_clause": ${newVal}`;
    });

    fs.writeFileSync(targetPath, updatedData, 'utf8');
    console.log("Successfully updated data_fighters.json release clauses.");
} catch (e) {
    console.error("Error patching data_fighters: ", e);
}
